services/base-jdbc/src/main/resources/org/eclipse/hono/service/base/jdbc/store/device/base.postgresql.sql.yaml

```diff
@@ -55,21 +55,75 @@ resolveGroups: |
       group_id in (select unnest((string_to_array(:group_ids,','))::varchar[]))
 
 countDevicesOfTenantWithFilter: |
-   SELECT COUNT(*) AS deviceCount FROM %1$s    
-   WHERE 
+   SELECT COUNT(*) AS deviceCount FROM %1$s
+   WHERE
       tenant_id=:tenant_id
    AND
      data->>:field LIKE CAST(:value AS VARCHAR) OR
      jsonb_extract_path(data,'ext')->>:field LIKE CAST(:value as varchar)
 
+countGatewaysOfTenant: |
+   SELECT COUNT(*) AS deviceCount FROM %1$s
+   WHERE
+      tenant_id=:tenant_id
+   AND
+      device_id IN (
+         SELECT replace(dr.device_id::text, '"', '') FROM (
+            SELECT jsonb_array_elements(data -> 'via') AS device_id FROM device_registrations 
+                  WHERE tenant_id=:tenant_id AND data ->> 'via' != '') 
+         as dr)
+
+countOnlyDevicesOfTenant: |
+   SELECT (SELECT COUNT(*) FROM %1$s WHERE tenant_id=:tenant_id)
+   -
+   (SELECT COUNT(*) AS deviceCount FROM %1$s
+   WHERE
+      tenant_id=:tenant_id
+   AND
+      device_id IN (
+         SELECT replace(dr.device_id::text, '"', '') FROM (
+            SELECT jsonb_array_elements(data -> 'via') AS device_id FROM device_registrations 
+                  WHERE tenant_id=:tenant_id AND data ->> 'via' != '') 
+         as dr)) AS deviceCount
+
 findDevicesOfTenantWithFilter: |
-   SELECT * 
-   FROM %s 
-   WHERE 
+   SELECT *
+   FROM %s
+   WHERE
```

Comment:
does this query work for a field like `/ext/my-struct/my-field`?

Reply:
yes, it should.